instrumentation/apache-dubbo-2.7/testing/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.java

```diff
@@ -0,0 +1,407 @@
+                                    NetworkAttributes.NETWORK_TYPE,
+                                    k ->
+                                        k.satisfiesAnyOf(
+                                            val -> assertThat(val).isNull(),
+                                            val -> assertThat(val).isEqualTo("ipv4"),
+                                            val -> assertThat(val).isEqualTo("ipv6")))),
+                    span ->
+                        span.hasName(
+                                "io.opentelemetry.instrumentation.apachedubbo.v2_7.api.HelloService/hello")
+                            .hasKind(SpanKind.SERVER)
+                            .hasAttributesSatisfying(
+                                equalTo(RpcIncubatingAttributes.RPC_SYSTEM, "apache_dubbo"),
+                                equalTo(
+                                    RpcIncubatingAttributes.RPC_SERVICE,
+                                    "io.opentelemetry.instrumentation.apachedubbo.v2_7.api.HelloService"),
+                                equalTo(RpcIncubatingAttributes.RPC_METHOD, "hello"),
+                                satisfies(
+                                    NetworkAttributes.NETWORK_PEER_ADDRESS,
+                                    k -> k.isInstanceOf(String.class)),
+                                satisfies(
+                                    NetworkAttributes.NETWORK_PEER_PORT,
+                                    k -> k.isInstanceOf(Long.class)),
+                                satisfies(
+                                    NetworkAttributes.NETWORK_TYPE,
+                                    k ->
+                                        k.satisfiesAnyOf(
+                                            val -> assertThat(val).isNull(),
+                                            val -> assertThat(val).isEqualTo("ipv4"),
+                                            val -> assertThat(val).isEqualTo("ipv6"))))));
+  }
+
+  @Test
+  @DisplayName("test ignore injvm calls")
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  void testDubboChainInJvm() {
+    int port = PortUtils.findOpenPorts(2);
+    int middlePort = port + 1;
+
+    // setup hello service provider
+    ProtocolConfig protocolConfig = new ProtocolConfig();
+    protocolConfig.setPort(port);
+
+    bootstrap = DubboTestUtil.newDubboBootstrap();
+    bootstrap
+        .application(new ApplicationConfig("dubbo-test-provider"))
+        .service(configureServer())
+        .protocol(protocolConfig)
+        .start();
+
+    // setup middle service provider, hello service consumer
+    ProtocolConfig middleProtocolConfig = new ProtocolConfig();
+    middleProtocolConfig.setPort(middlePort);
+
+    ReferenceConfig<HelloService> clientReference = configureLocalClient(port);
+    middleBootstrap = DubboTestUtil.newDubboBootstrap();
+    middleBootstrap
+        .application(new ApplicationConfig("dubbo-demo-middle"))
+        //        .reference(clientReference)
```

Comment:
should this be commented out?

Reply:
Yes, the method must be commented.

It is wrong usage for difference interface in the same bootstrap. It will cause some error.

Reply:
> Yes, the method must be commented.
> 
> It is wrong usage for difference interface in the same bootstrap. It will cause some error.

Is it better to delete useless code instead of commenting?